example for injecting cmake variables (#3333)
conan-io/docs: examples/tools/cmake/cmake_toolchain/inject_cmake_variables.rst

.. _examples-tools-cmake-toolchain-inject-variables:

CMakeToolchain: Inject arbitrary CMake variables into dependencies
==================================================================

> In the general case, Conan package recipe provides the necessary abstractions via settings, confs, and options

Reviewer: ```suggestion
In the general case, Conan package recipes provide the necessary abstractions via settings, confs, and options
```

to control different aspects of the build. Many recipes define ``options`` to activate or deactivate features,
optional dependencies, or binary characteristics. Configurations like ``tools.build:cxxflags`` can be used to
inject arbitrary C++ compile flags.

In some exceptional cases, it might be desired to inject CMake variables directly into dependencies doing CMake

> builds. This is possible when these dependencies use the ``CMakeToolchan`` integration. Let's check it in this

Reviewer 2: ```suggestion
builds. This is possible when these dependencies use the ``CMakeToolchain`` integration. Let's check it in this
```

.. _examples-tools-cmake-toolchain-inject-variables:

CMakeToolchain: Inject arbitrary CMake variables into dependencies
==================================================================

> 

Reviewer 2: ```suggestion

You can find the sources to recreate this project in the `examples2.0 repository
<https://github.com/conan-io/examples2>`_ in GitHub:

.. code-block:: bash

    $ git clone https://github.com/conan-io/examples2.git
    $ cd examples2/examples/tools/cmake/cmake_toolchain/user_toolchain_profile

```
